utils/analysis/filter.go

```diff
@@ -148,6 +163,19 @@ func FilterAnalysisRunsToDelete(ars []*v1alpha1.AnalysisRun, olderRSs []*appsv1.
 			arsToDelete = append(arsToDelete, ar)
 			continue
 		}
+
+		if ar.Status.Phase == v1alpha1.AnalysisPhaseSuccessful {
+			if retainedSucceed < limitSucceedArs {
+				retainedSucceed++
+				continue
+			}
+		} else {
```

Comment:
Shall we distinguish the analysis status between Failed and Running?

Reply:
Here we're filtering the `otherArs`. As for my understanding, the other (means old) are finished already so they can have succeeded or failed state.

So here I'm checking for 2 subsets of ARs (`ar.Status.Phase == v1alpha1.AnalysisPhaseSuccessful`) and `else` means failed or error.

@huikang What do you think, is that OK or I have to check explicitly for the error/failed in an `else` block?